### PR TITLE
[PF-1036] Implementation and test of application configuration

### PR DESCRIPTION
**Pull request by ddietterich**
This is the first part of implementing applications. It doesn't change any external behavior of the WSM.

### CODE REVIEW

File: service/src/test/java/bio/terra/workspace/service/workspace/ApplicationUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    appService.enableTestMode();
+    wsmApp.displayName(GOOD_NAME);
+    wsmApp.description(GOOD_DESC);
+    appService.processApp(wsmApp, dbAppMap);
+    assertMessage(0, INFO_UPDATED);
+
+    // -- 4th pass -- State transition to disabled
+    stateTransition(WsmApplicationState.DISABLED, INFO_UPDATED);
+
+    // -- 5th pass -- State transition bak to enabled
```
Comment (zloery) [inline service/src/test/java/bio/terra/workspace/service/workspace/ApplicationUnitTest.java:151]: back*

File: service/src/main/resources/db/changesets/20211006_application_columns.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    author: dd
+    changes:
+    - addColumn:
+        tableName: application
+        columns:
+          - column:
+              name: state
+              type: text
+              remarks: |
+                From WsmApplicationState: disabled, enabled, decommissioned
```
Comment (zloery) [inline service/src/main/resources/db/changesets/20211006_application_columns.yaml:13]: deprecated, operating, decomissioned

**Review by zloery — APPROVED** [review]
LGTM!

File: service/src/test/java/bio/terra/workspace/service/workspace/ApplicationUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+    assertMessage(1, ERROR_DUPLICATE);
+
+    // -- 3rd pass -- update name and desc
+    dbAppMap = appService.buildAppMap();
+    appService.enableTestMode();
+    wsmApp.displayName(GOOD_NAME);
+    wsmApp.description(GOOD_DESC);
+    appService.processApp(wsmApp, dbAppMap);
+    assertMessage(0, INFO_UPDATED);
+
```
Comment (jmczerk) [inline service/src/test/java/bio/terra/workspace/service/workspace/ApplicationUnitTest.java:147]: `s/disabled/deprecated`
`s/enabled/operating`

**Review by jmczerk — APPROVED** [review]

**Pull request merged**